internal/spoa.go

```diff
@@ -229,3 +229,169 @@ func New(conf *config.Config) (*SPOA, error) {
 		defaultApplication: conf.DefaultApplication,
 	}, nil
 }
+
+func (s *SPOA) getApplication(appName string) (*application, error) {
+	var app *application
+
+	// Looking for app by name from message
+	if appName != "" {
+		app, exist := s.applications[appName]
+		if exist {
+			return app, nil
+		}
+	}
+
+	// Looking for app by default app name
+	app, exist := s.applications[s.defaultApplication]
+	if exist {
+		app.logger.Debug("application not found, using default", zap.Any("application", appName), zap.String("default", s.defaultApplication))
+		return app, nil
+	}
+
+	return nil, fmt.Errorf("application not found, application %s, default: %s", appName, s.defaultApplication)
+}
+
+func (s *SPOA) processRequest(spoeMsg spoe.Message) ([]spoe.Action, error) {
+	var (
+		err error
+		req *request
+		app *application
+		tx  types.Transaction
+	)
+
+	defer func() {
+		if tx == nil || app == nil {
+			return
+		}
+		if tx.IsInterrupted() {
+			tx.ProcessLogging()
+			if err := tx.Close(); err != nil {
+				app.logger.Error("failed to close transaction", zap.String("transaction_id", tx.ID()), zap.String("error", err.Error()))
+			}
+		} else {
+			if app.cfg.NoResponseCheck {
+				return
+			}
+			err := app.cache.SetWithExpire(tx.ID(), tx, time.Millisecond*time.Duration(app.cfg.TransactionTTLMilliseconds))
+			if err != nil {
+				app.logger.Error(fmt.Sprintf("failed to cache transaction: %s", err.Error()))
+			}
+		}
+	}()
+
+	req, err = NewRequest(spoeMsg)
+	if err != nil {
+		return nil, err
+	}
+
+	app, err = s.getApplication(req.app)
+	if err != nil {
+		return nil, err
+	}
+
+	tx = app.waf.NewTransactionWithID(req.id)
+	if tx.IsRuleEngineOff() {
```

Comment:
Do we really need to execute all the previous code if ruleEngine is off?

Reply:
We need [application at least](https://github.com/zc-devs/coraza-spoa/blob/d1a3d83894ea675ed2709ceb5c8ae5b35ee4a6e1/internal/spoa.go#L198-L209). Then we could get [rule engine status from WAF](https://github.com/corazawaf/coraza/blob/5ffc3a359c8ebf3f0890aa845e611834669066f6/internal/corazawaf/waf.go#L57), if it would be public interface. [In public interface there are only transactions constructors](https://github.com/corazawaf/coraza/blob/5ffc3a359c8ebf3f0890aa845e611834669066f6/waf.go#L21-L25). So, I think we need  to execute all previous code.